static/app/components/sidebar/broadcasts.tsx

```diff
@@ -28,15 +28,77 @@ type Props = CommonSidebarProps & {
 
 type State = {
   broadcasts: Broadcast[];
-  error: boolean;
   loading: boolean;
 };
 
+function BroadcastSidebarContent({
+  orientation,
+  collapsed,
+  loading,
+  broadcasts,
+  whatIsNewRevampFeature,
+  hidePanel,
+  onResetCounter,
+}: {
+  broadcasts: Broadcast[];
+  loading: boolean;
+  onResetCounter: () => void;
+  whatIsNewRevampFeature: boolean;
+} & Pick<CommonSidebarProps, 'orientation' | 'collapsed' | 'hidePanel'>) {
+  useEffect(() => {
+    return () => {
+      if (whatIsNewRevampFeature) {
+        onResetCounter();
+      }
+    };
+  }, [onResetCounter, whatIsNewRevampFeature]);
```

Comment:
Nit: Why extract all this into a separate component and not just do it in `componentWillUnmount`?

Reply:
nvm, just figured it out. 🙈 We should migrate the class component to a functional one...

Reply:
yes... i have a code prepared 🙂